services/src/datasets/listing.rs

```diff
@@ -0,0 +1,120 @@
+use crate::datasets::storage::{AddDataSetProvider, AddMockDataSetProvider, DataSet};
+use crate::error::Result;
+use crate::users::user::UserId;
+use crate::util::user_input::{UserInput, Validated};
+use async_trait::async_trait;
+use geoengine_datatypes::dataset::DataSetId;
+use geoengine_operators::engine::{
+    MetaData, MetaDataProvider, ResultDescriptor, VectorResultDescriptor,
+};
+use geoengine_operators::mock::MockDataSetDataSourceLoadingInfo;
+use geoengine_operators::source::OgrSourceDataset;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
+pub struct DataSetListing {
+    pub id: DataSetId,
+    pub name: String,
+    pub description: String,
+    pub tags: Vec<String>,
+    pub source_operator: String, // TODO: enum?
+                                 // TODO: meta data like bounds, crs, resolution
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct DataSetListOptions {
+    // TODO: permissions
+    pub filter: Option<DataSetFilter>,
+    pub order: OrderBy,
+    pub offset: u32,
+    pub limit: u32,
+}
+
+impl UserInput for DataSetListOptions {
+    fn validate(&self) -> Result<()> {
+        // TODO
+        Ok(())
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct DataSetFilter {
+    pub name: String,
+    // TODO: tags, ..
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Clone, Hash)]
+pub enum OrderBy {
+    NameAsc,
+    NameDesc,
+}
```

Comment:
I guess we have or will have redundant ordering enums. Do you think we should merge them?

Reply:
Merging is difficult, because different types have different attributes to sort by. We would need something like
```
struct Ordering {
  attribute: Attribute,
  order: Order,
}
```
but ideally it should still serialize to a string. I guess it's more of an "Idea" task